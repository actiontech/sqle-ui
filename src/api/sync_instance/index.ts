/* tslint:disable no-identical-functions */
/* tslint:disable no-useless-cast */
/* tslint:disable no-unnecessary-type-assertion */
/* tslint:disable no-big-function  */
/* tslint:disable no-duplicate-string  */
import ServiceBase from '../Service.base';
import { AxiosRequestConfig } from 'axios';

import {
  IGetSyncInstanceTaskListReturn,
  ICreateSyncInstanceTaskV1Params,
  ICreateSyncInstanceTaskV1Return,
  IGetSyncTaskSourceTipsReturn,
  IGetSyncInstanceTaskParams,
  IGetSyncInstanceTaskReturn,
  IDeleteSyncInstanceTaskV1Params,
  IDeleteSyncInstanceTaskV1Return,
  IUpdateSyncInstanceTaskV1Params,
  IUpdateSyncInstanceTaskV1Return,
  ITriggerSyncInstanceV1Params,
  ITriggerSyncInstanceV1Return
} from './index.d';

class SyncInstanceService extends ServiceBase {
  public GetSyncInstanceTaskList(options?: AxiosRequestConfig) {
    return this.get<IGetSyncInstanceTaskListReturn>(
      '/v1/sync_instance',
      undefined,
      options
    );
  }

  public createSyncInstanceTaskV1(
    params: ICreateSyncInstanceTaskV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    return this.post<ICreateSyncInstanceTaskV1Return>(
      '/v1/sync_instance',
      paramsData,
      options
    );
  }

  public GetSyncTaskSourceTips(options?: AxiosRequestConfig) {
    return this.get<IGetSyncTaskSourceTipsReturn>(
      '/v1/sync_instance/source_tips',
      undefined,
      options
    );
  }

  public GetSyncInstanceTask(
    params: IGetSyncInstanceTaskParams,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    const task_id = paramsData.task_id;
    delete paramsData.task_id;

    return this.get<IGetSyncInstanceTaskReturn>(
      `/v1/sync_instance/${task_id}/`,
      paramsData,
      options
    );
  }

  public deleteSyncInstanceTaskV1(
    params: IDeleteSyncInstanceTaskV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    const task_id = paramsData.task_id;
    delete paramsData.task_id;

    return this.delete<IDeleteSyncInstanceTaskV1Return>(
      `/v1/sync_instance/${task_id}/`,
      paramsData,
      options
    );
  }

  public updateSyncInstanceTaskV1(
    params: IUpdateSyncInstanceTaskV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    const task_id = paramsData.task_id;
    delete paramsData.task_id;

    return this.patch<IUpdateSyncInstanceTaskV1Return>(
      `/v1/sync_instance/${task_id}/`,
      paramsData,
      options
    );
  }

  public triggerSyncInstanceV1(
    params: ITriggerSyncInstanceV1Params,
    options?: AxiosRequestConfig
  ) {
    const paramsData = this.cloneDeep(params);
    const task_id = paramsData.task_id;
    delete paramsData.task_id;

    return this.post<ITriggerSyncInstanceV1Return>(
      `/v1/sync_instance/${task_id}/trigger`,
      paramsData,
      options
    );
  }
}

export default new SyncInstanceService();
