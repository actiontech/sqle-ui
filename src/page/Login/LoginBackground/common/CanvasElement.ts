import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
