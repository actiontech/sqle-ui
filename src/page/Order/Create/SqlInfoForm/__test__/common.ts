import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
