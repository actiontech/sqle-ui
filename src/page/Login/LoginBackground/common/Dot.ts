import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
