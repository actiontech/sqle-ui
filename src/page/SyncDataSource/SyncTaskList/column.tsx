import { Divider, Popconfirm, Space, Tag, Typography } from 'antd';
import { Link } from 'react-router-dom';
import { IInstanceTaskResV1 } from '../../../api/common';
import { InstanceTaskResV1LastSyncStatusEnum } from '../../../api/common.enum';
import i18n from '../../../locale';
import { TableColumn } from '../../../types/common.type';
import { formatTime } from '../../../utils/Common';

export const SyncTaskListTableColumnFactory: (
  syncAction: (taskId: string) => void,
  deleteAction: (taskId: string) => void
) => TableColumn<IInstanceTaskResV1, 'operator'> = (
  syncAction,
  deleteAction
) => {
  return [
    {
      dataIndex: 'source',
      title: () => i18n.t('syncDataSource.syncTaskList.columns.source'),
    },
    {
      dataIndex: 'version',
      title: () => i18n.t('syncDataSource.syncTaskList.columns.version'),
    },
    {
      dataIndex: 'url',
      title: () => i18n.t('syncDataSource.syncTaskList.columns.url'),
    },
    {
      dataIndex: 'db_type',
      title: () => i18n.t('syncDataSource.syncTaskList.columns.instanceType'),
    },
    {
      dataIndex: 'last_sync_status',
      title: () => i18n.t('syncDataSource.syncTaskList.columns.lastSyncResult'),
      render: (status: InstanceTaskResV1LastSyncStatusEnum) => {
        if (status === InstanceTaskResV1LastSyncStatusEnum.success) {
          return <Tag color="green">{i18n.t('common.success')}</Tag>;
        } else if (status === InstanceTaskResV1LastSyncStatusEnum.fail) {
          return <Tag color="red">{i18n.t('common.fail')}</Tag>;
        }
      },
    },
    {
      dataIndex: 'last_sync_success_time',
      title: () =>
        i18n.t('syncDataSource.syncTaskList.columns.lastSyncSuccessTime'),
      render: (time) => {
        return formatTime(time);
      },
    },
    {
      dataIndex: 'operator',
      title: i18n.t('common.operate'),
      width: 160,
      render: (_, record) => {
        return (
          <Space className="user-cell flex-end-horizontal">
            <Link to={`/syncDataSource/update/${record.id?.toString()}`}>
              {i18n.t('common.edit')}
            </Link>
            <Typography.Link
              className="pointer"
              onClick={syncAction.bind(null, record.id?.toString() ?? '')}
            >
              {i18n.t('syncDataSource.syncTaskList.columns.sync')}
            </Typography.Link>
            <Divider type="vertical" />
            <Popconfirm
              title={i18n.t(
                'syncDataSource.syncTaskList.columns.deleteConfirmTitle'
              )}
              placement="topRight"
              okText={i18n.t('common.ok')}
              cancelText={i18n.t('common.cancel')}
              onConfirm={deleteAction.bind(null, record.id?.toString() ?? '')}
            >
              <Typography.Text type="danger" className="pointer">
                {i18n.t('common.delete')}
              </Typography.Text>
            </Popconfirm>
          </Space>
        );
      },
    },
  ];
};
