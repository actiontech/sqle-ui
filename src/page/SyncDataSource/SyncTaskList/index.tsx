import { SyncOutlined } from '@ant-design/icons';
import { useRequest } from 'ahooks';
import { Button, Card, message, Space, Table } from 'antd';
import { useTranslation } from 'react-i18next';
import sync_instance from '../../../api/sync_instance';
import { ResponseCode } from '../../../data/common';
import { SyncTaskListTableColumnFactory } from './column';
import { Link } from '../../../components/Link';

const SyncTaskList: React.FC = () => {
  const { t } = useTranslation();

  const syncAction = (taskId: string) => {
    const hideLoading = message.loading(
      t('syncDataSource.syncTaskList.syncTaskLoading')
    );
    sync_instance
      .triggerSyncInstanceV1({
        task_id: taskId,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          message.success(t('syncDataSource.syncTaskList.syncTaskSuccessTips'));
          refresh();
        }
      })
      .finally(() => {
        hideLoading();
      });
  };
  const deleteAction = (taskId: string) => {
    const hideLoading = message.loading(
      t('syncDataSource.syncTaskList.deleteTaskLoading')
    );
    sync_instance
      .deleteSyncInstanceTaskV1({
        task_id: taskId,
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          message.success(
            t('syncDataSource.syncTaskList.deleteTaskSuccessTips')
          );
          refresh();
        }
      })
      .finally(() => {
        hideLoading();
      });
  };

  const { loading, data, refresh } = useRequest(() =>
    sync_instance
      .GetSyncInstanceTaskList()
      .then((res) => ({ list: res.data?.data ?? [] }))
  );

  return (
    <Card
      title={
        <Space>
          {t('syncDataSource.syncTaskList.title')}
          <Button onClick={refresh} data-testid="refreshTable">
            <SyncOutlined spin={loading} />
          </Button>
        </Space>
      }
      extra={[
        <Button type="primary" key="addSyncTask">
          <Link to="syncDataSource/create">
            {t('syncDataSource.syncTaskList.addSyncTask')}
          </Link>
        </Button>,
      ]}
    >
      <Table
        rowKey="id"
        dataSource={data?.list ?? []}
        loading={loading}
        pagination={false}
        columns={SyncTaskListTableColumnFactory(syncAction, deleteAction)}
      />
    </Card>
  );
};

export default SyncTaskList;
