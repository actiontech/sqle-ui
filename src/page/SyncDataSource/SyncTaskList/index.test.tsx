import { fireEvent, screen, waitFor } from '@testing-library/react';
import SyncTaskList from '.';
import { IInstanceTaskResV1 } from '../../../api/common';
import { InstanceTaskResV1LastSyncStatusEnum } from '../../../api/common.enum';
import sync_instance from '../../../api/sync_instance';
import {
  renderWithRouter,
  renderWithServerRouter,
} from '../../../testUtils/customRender';
import { resolveThreeSecond } from '../../../testUtils/mockRequest';
import { createMemoryHistory } from 'history';

const tableList: IInstanceTaskResV1[] = [
  {
    db_type: 'mysql',
    id: 0,
    last_sync_status: InstanceTaskResV1LastSyncStatusEnum.success,
    last_sync_success_time: '2023-01-10',
    source: 'source1',
    url: 'http://192.168.1.1:3000',
    version: '4.2.2.0',
  },
  {
    db_type: 'oracle',
    id: 1,
    last_sync_status: InstanceTaskResV1LastSyncStatusEnum.fail,
    last_sync_success_time: '2023-01-10',
    source: 'source1',
    url: 'http://192.168.1.1:3000',
    version: '4.2.2.0',
  },
];

describe('test SyncTaskList', () => {
  let getSyncInstanceTaskListSpy: jest.SpyInstance;
  let triggerSyncInstanceSpy: jest.SpyInstance;
  let deleteSyncInstanceTaskSpy: jest.SpyInstance;
  beforeEach(() => {
    getSyncInstanceTaskListSpy = mockGetSyncInstanceTaskList();
    triggerSyncInstanceSpy = mockTriggerSyncInstance();
    deleteSyncInstanceTaskSpy = mockDeleteSyncInstanceTask();
    jest.useFakeTimers();
  });
  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  const mockGetSyncInstanceTaskList = () => {
    const spy = jest.spyOn(sync_instance, 'GetSyncInstanceTaskList');
    spy.mockImplementation(() => resolveThreeSecond(tableList));
    return spy;
  };
  const mockTriggerSyncInstance = () => {
    const spy = jest.spyOn(sync_instance, 'triggerSyncInstanceV1');
    spy.mockImplementation(() => resolveThreeSecond({}));
    return spy;
  };

  const mockDeleteSyncInstanceTask = () => {
    const spy = jest.spyOn(sync_instance, 'deleteSyncInstanceTaskV1');
    spy.mockImplementation(() => resolveThreeSecond({}));
    return spy;
  };

  test('should match snapshot', () => {
    const { container } = renderWithRouter(<SyncTaskList />);

    expect(container).toMatchSnapshot();
  });

  test('should be called request to get data for the render table', async () => {
    expect(getSyncInstanceTaskListSpy).toBeCalledTimes(0);
    const { container } = renderWithRouter(<SyncTaskList />);
    expect(getSyncInstanceTaskListSpy).toBeCalledTimes(1);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(container).toMatchSnapshot();

    fireEvent.click(screen.getByTestId('refreshTable'));
    expect(getSyncInstanceTaskListSpy).toBeCalledTimes(2);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
  });

  test('should jump to "/syncDataSource/create" when clicking add sync task button', () => {
    const history = createMemoryHistory();
    renderWithServerRouter(<SyncTaskList />, undefined, { history });
    expect(history.location.pathname).toBe('/');

    fireEvent.click(
      screen.getByText('syncDataSource.syncTaskList.addSyncTask')
    );
    expect(history.location.pathname).toBe('/syncDataSource/create');
  });

  test('should jump to "/syncDataSource/update" when clicking edit sync task button', async () => {
    const history = createMemoryHistory();
    renderWithServerRouter(<SyncTaskList />, undefined, { history });
    expect(history.location.pathname).toBe('/');

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });

    expect(screen.queryAllByText('common.edit').length).toBe(2);

    fireEvent.click(screen.queryAllByText('common.edit')[0]);

    expect(history.location.pathname).toBe('/syncDataSource/update/0');

    fireEvent.click(screen.queryAllByText('common.edit')[1]);

    expect(history.location.pathname).toBe('/syncDataSource/update/1');
  });

  test('should be send sync request when clicking sync task button', async () => {
    expect(triggerSyncInstanceSpy).toBeCalledTimes(0);
    renderWithRouter(<SyncTaskList />);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.click(
      screen.queryAllByText('syncDataSource.syncTaskList.columns.sync')[0]
    );
    expect(triggerSyncInstanceSpy).toBeCalledTimes(1);
    expect(triggerSyncInstanceSpy).toBeCalledWith({
      task_id: '0',
    });
    expect(
      screen.queryByText('syncDataSource.syncTaskList.syncTaskLoading')
    ).toBeInTheDocument();
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.queryByText('syncDataSource.syncTaskList.syncTaskLoading')
    ).not.toBeInTheDocument();

    expect(
      screen.queryByText('syncDataSource.syncTaskList.syncTaskSuccessTips')
    ).toBeInTheDocument();

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.queryByText('syncDataSource.syncTaskList.syncTaskSuccessTips')
    ).not.toBeInTheDocument();
  });

  test('should be send delete request when clicking delete task button', async () => {
    expect(deleteSyncInstanceTaskSpy).toBeCalledTimes(0);
    renderWithRouter(<SyncTaskList />);
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    fireEvent.click(screen.queryAllByText('common.delete')[0]);
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(
      screen.queryByText(
        'syncDataSource.syncTaskList.columns.deleteConfirmTitle'
      )
    ).toBeInTheDocument();

    fireEvent.click(screen.getByText('common.ok'));

    expect(deleteSyncInstanceTaskSpy).toBeCalledTimes(1);
    expect(deleteSyncInstanceTaskSpy).toBeCalledWith({
      task_id: '0',
    });
    expect(
      screen.queryByText('syncDataSource.syncTaskList.deleteTaskLoading')
    ).toBeInTheDocument();
    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.queryByText('syncDataSource.syncTaskList.deleteTaskLoading')
    ).not.toBeInTheDocument();

    expect(
      screen.queryByText('syncDataSource.syncTaskList.deleteTaskSuccessTips')
    ).toBeInTheDocument();

    await waitFor(() => {
      jest.advanceTimersByTime(3000);
    });
    expect(
      screen.queryByText('syncDataSource.syncTaskList.deleteTaskSuccessTips')
    ).not.toBeInTheDocument();
  });
});
