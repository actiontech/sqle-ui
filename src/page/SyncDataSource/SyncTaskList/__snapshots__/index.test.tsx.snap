// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SyncTaskList should be called request to get data for the render table 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 8px;"
            >
              syncDataSource.syncTaskList.title
            </div>
            <div
              class="ant-space-item"
            >
              <button
                class="ant-btn"
                data-testid="refreshTable"
                type="button"
              >
                <span
                  aria-label="sync"
                  class="anticon anticon-sync"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="sync"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="ant-card-extra"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <a
              href="/syncDataSource/create"
            >
              syncDataSource.syncTaskList.addSyncTask
            </a>
          </button>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col
                        style="width: 160px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.source
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.version
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.url
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.instanceType
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.lastSyncResult
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.lastSyncSuccessTime
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          common.operate
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="0"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          source1
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          4.2.2.0
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          http://192.168.1.1:3000
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          mysql
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-tag ant-tag-green"
                          >
                            common.success
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2023-01-10 00:00:00
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center user-cell flex-end-horizontal"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <a
                                href="/syncDataSource/update/0"
                              >
                                common.edit
                              </a>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <a
                                class="ant-typography pointer"
                              >
                                syncDataSource.syncTaskList.columns.sync
                              </a>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <div
                                class="ant-divider ant-divider-vertical"
                                role="separator"
                              />
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <span
                                class="ant-typography ant-typography-danger pointer"
                              >
                                common.delete
                              </span>
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          source1
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          4.2.2.0
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          http://192.168.1.1:3000
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          oracle
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            class="ant-tag ant-tag-red"
                          >
                            common.fail
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          2023-01-10 00:00:00
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center user-cell flex-end-horizontal"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <a
                                href="/syncDataSource/update/1"
                              >
                                common.edit
                              </a>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <a
                                class="ant-typography pointer"
                              >
                                syncDataSource.syncTaskList.columns.sync
                              </a>
                            </div>
                            <div
                              class="ant-space-item"
                              style="margin-right: 8px;"
                            >
                              <div
                                class="ant-divider ant-divider-vertical"
                                role="separator"
                              />
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <span
                                class="ant-typography ant-typography-danger pointer"
                              >
                                common.delete
                              </span>
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test SyncTaskList should match snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 8px;"
            >
              syncDataSource.syncTaskList.title
            </div>
            <div
              class="ant-space-item"
            >
              <button
                class="ant-btn"
                data-testid="refreshTable"
                type="button"
              >
                <span
                  aria-label="sync"
                  class="anticon anticon-sync anticon-spin"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="sync"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="ant-card-extra"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <a
              href="/syncDataSource/create"
            >
              syncDataSource.syncTaskList.addSyncTask
            </a>
          </button>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-table-wrapper"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div>
            <div
              class="ant-spin ant-spin-spinning"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-spin-container ant-spin-blur"
          >
            <div
              class="ant-table ant-table-empty"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col />
                      <col
                        style="width: 160px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.source
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.version
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.url
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.instanceType
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.lastSyncResult
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          syncDataSource.syncTaskList.columns.lastSyncSuccessTime
                        </th>
                        <th
                          class="ant-table-cell"
                        >
                          common.operate
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-placeholder"
                      >
                        <td
                          class="ant-table-cell"
                          colspan="7"
                        >
                          <div
                            class="ant-empty ant-empty-normal"
                          >
                            <div
                              class="ant-empty-image"
                            >
                              <svg
                                class="ant-empty-img-simple"
                                height="41"
                                viewBox="0 0 64 41"
                                width="64"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  fill="none"
                                  fill-rule="evenodd"
                                  transform="translate(0 1)"
                                >
                                  <ellipse
                                    class="ant-empty-img-simple-ellipse"
                                    cx="32"
                                    cy="33"
                                    rx="32"
                                    ry="7"
                                  />
                                  <g
                                    class="ant-empty-img-simple-g"
                                    fill-rule="nonzero"
                                  >
                                    <path
                                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                    />
                                    <path
                                      class="ant-empty-img-simple-path"
                                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </div>
                            <div
                              class="ant-empty-description"
                            >
                              No Data
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
